start.py

```diff
@@ -48,35 +56,54 @@ def main(args):
     secondary_nodes = [Node(hostname=hostname, group='secondary', image=secondary_node_image)
                        for hostname in args.secondary_nodes]
 
-    cluster = Cluster(primary_node, *secondary_nodes)
+    cluster = Cluster(primary_node, *secondary_nodes, name=args.cluster_name)
     cluster.primary_node = primary_node
     cluster.secondary_nodes = secondary_nodes
     cluster.start(args.network)
 
+    hdp_version_tuple = version_tuple(args.hdp_version)
+
     logger.debug('Starting PostgreSQL for Ambari server ...')
-    primary_node.execute('service postgresql start', quiet=not args.verbose)
-    _update_node_names(cluster, quiet=not args.verbose)
+    # Need this as init system in Docker misreports on postgres start initially
+    # Check https://github.com/docker-library/postgres/issues/146 for more
+    def condition():
+        if '1 row' not in primary_node.execute('PGPASSWORD=bigdata psql ambari '
+                                               '-U ambari -h localhost -c "select 1"',
+                                               quiet=quiet).output:
+            primary_node.execute('service postgresql restart', quiet=quiet)
+            time.sleep(1)
+            return False
+        return True
+    wait_for_condition(condition=condition)
+
+    def condition():
```

Comment:
Same comment as above about being able to remove the `return False` by reexpressing this condition